cla_backend/apps/cla_eventlog/tests/test_find_and_delete_old_cases.py

```diff
@@ -0,0 +1,72 @@
+from django.test import TestCase
+from django.utils import timezone
+from dateutil.relativedelta import relativedelta
+from freezegun import freeze_time
+
+from cla_butler.tasks import get_pks
+from core.tests.mommy_utils import make_recipe
+from cla_eventlog.models import Log
+from legalaid.models import Case
+
+
+class FindAndDeleteOldCases(TestCase):
+    def create_three_year_old_case(self):
+        case = None
+        freezer = freeze_time(timezone.now() + relativedelta(years=-3))
+        freezer.start()
+        case = make_recipe("legalaid.case")
+        freezer.stop()
+        return case
+
+    def create_event_log_for_case(self, case, code, created=None):
+        if created:
+            make_recipe("cla_eventlog.log", case=case, code=code, created=created)
+        else:
+            make_recipe("cla_eventlog.log", case=case, code=code)
+
+    def test_find_cases_viewed_two_years_ago(self):
+        new_case = make_recipe("legalaid.case")
+        old_case_1 = self.create_three_year_old_case()
+        old_case_2 = self.create_three_year_old_case()
+
+        self.assertEqual(len(Case.objects.all()), 3)
+
+        self.create_event_log_for_case(new_case, "CASE_VIEWED")
+        self.create_event_log_for_case(new_case, "CASE_CREATED")
+        self.create_event_log_for_case(new_case, "CASE_VIEWED", created=timezone.now() + relativedelta(years=-3))
+
+        self.create_event_log_for_case(old_case_1, "CASE_VIEWED")
+        self.create_event_log_for_case(old_case_1, "MT_CHANGED")
+        self.create_event_log_for_case(old_case_1, "CASE_VIEWED", created=timezone.now() + relativedelta(years=-1))
+
+        self.create_event_log_for_case(old_case_2, "CASE_VIEWED", created=timezone.now() + relativedelta(years=-3))
+        self.create_event_log_for_case(old_case_2, "CASE_CREATED", created=timezone.now() + relativedelta(years=-3))
+        self.create_event_log_for_case(old_case_2, "CASE_VIEWED", created=timezone.now() + relativedelta(years=-2))
+
+        logs = Log.objects.all()
+
+        self.assertEqual(len(logs), 9)
+
+        case_pks = get_pks(Case.objects.all())
+        cases_and_logs = {}
```

Comment:
Why do we need a dict? What part of your code does it facilitate?

Reply:
It's so I can store the all the logs for each case as cases and logs have no direct relationship.

Did it this way to fix the previous issue of finding logs that were over 2 years ago, but not knowing whether the same case had any newer logs.

Reply:
Fair enough. But you only write to it once per case and you consume it by running through it as a list of tuples (using .items()) so it could just be a list of tuples. Anyway this may well be moot if you end up restructuring at all

Reply:
Ah ok. Most likely I won't be restructuring, rather just reusing the data already there. I'll change it into a tuple

Reply:
I meant restructuring your code. You may end up not needing an intermediary data structure at all in the end, but while you do I would use the simplest that does what you need

Reply:
Uh I edited my comment, but commented too late.. but cool. When I move this into a function, I'll be able to restructure there

Reply:
> Yeah since I write to it, I can't make it into a tuple. Unless you mean to convert it afterwards, then sure.
I didn't understand what you meant but I do now. Not a tuple. A list of tuples. The list will be mutable, the inner ones don't need to be.

Reply:
Oh yikes I read that wrong. We've since changed it, so it no longer uses a dict. Instead we filter through cases that are less than 2 years old, then filter through the logs based on the cases.